File: erpnext_gsg/public/js/report_sales_order.js
frappe.provide('erpnext.accounts.report');

erpnext.accounts.report.SalesOrderAnalysis = erpnext.accounts.report.SalesOrderAnalysis.extend({
    get_filters: function () {
        var filters = this._super();
        filters.from_time = this.$wrapper.find('[name="from_time"]').val();
        filters.to_time = this.$wrapper.find('[name="to_time"]').val();
        return filters;
    }
});
frappe.ui.form.on('Sales Order', {
    onload: function (frm) {
        frm.set_query('sales_order_time', function () {
            return {
                filters: {
                    'docstatus': 1
                }
            };
        });
    }
});

File: erpnext_gsg/erp_next_gsg/doc_event/report_sales_order.py
import frappe
from frappe import _
from erpnext.accounts.report.sales_order_analysis.sales_order_analysis import SalesOrderAnalysis

def sales_order_analysis_filter(self):
    filters = self.get_filters()
    from_time = filters.get('from_time')
    to_time = filters.get('to_time')

    if not from_time:
        frappe.throw(_('From Time is mandatory'))

    if not to_time:
        frappe.throw(_('To Time is mandatory'))

    return {'from_time': from_time, 'to_time': to_time}

SalesOrderAnalysis.get_filters = sales_order_analysis_filter














